ITE: soc: Add the variants of it82xx2bw

GTLin08 opened the pull request:
This PR includes five commits, 
1. Using the SOC_IT8XXX2_REG_SET_V2 instead of constantly adding new variants of the IT82XX2 SoC.
2. Remove underscore makes config name consistent.
3. Add the description to migration-guide-3.7 and release-notes-3.7
4. Adding it82202bw variant.
5. Adding it82302bw variant.

GTLin08 committed:
ITE: soc: Add the variant of it82202bw

Add the variant of it82202bw.

Signed-off-by: Tim Lin <tim2.lin@ite.corp-partner.google.com>

GTLin08 committed:
ITE: soc: Add the variant of it82302bw

Add the variant of it82302bw.

Signed-off-by: Tim Lin <tim2.lin@ite.corp-partner.google.com>

nordicjm commented on soc/ite/ec/it8xxx2/Kconfig.soc:
```diff
@@ -50,3 +54,4 @@ config SOC
 	default "it82002aw" if SOC_IT82002_AW
 	default "it82202ax" if SOC_IT82202_AX
 	default "it82302ax" if SOC_IT82302_AX
+	default "it82202bw" if SOC_IT82202_BW
```
I did not notice before, why is there an underscore in these Kconfig symbols? The names here should match e.g. `default "it82202_bw" if SOC_IT82202_BW` or `default "it82202bw" if SOC_IT82202BW`

GTLin08 replied:
Remove underscore.

nordicjm commented on soc/ite/ec/it8xxx2/Kconfig.soc:
```diff
@@ -39,6 +39,10 @@ config SOC_IT82002_AW
+	select SOC_IT8XXX2
+
 config SOC_SERIES
 	default "it8xxx2" if SOC_SERIES_ITE_IT8XXX2
```
Same with this, the names should match e.g. `default "it8xxx2" if SOC_SERIES_IT8XXX2` or `default "ite_it8xxx2" if SOC_SERIES_ITE_IT8XXX2`

GTLin08 replied:
Remove "_ITE".

GTLin08 committed:
ITE: soc: Kconfig: Remove underscore makes config names consistent

Remove underscore makes config name consistent.

Signed-off-by: Tim Lin <tim2.lin@ite.corp-partner.google.com>

keith-zephyr commented on boards/ite/it82xx2_evb/Kconfig.it82xx2_evb:
```diff
@@ -2,4 +2,4 @@
 
 config BOARD_IT82XX2_EVB
-	select SOC_IT82202_AX
+	select SOC_IT82202AX
```
You need to document this change in the release notes and migration guide.

GTLin08 replied:
Done.

GTLin08 committed:
doc: migration-guide-3.7 & release-notes-3.7: Rename the Kconfig symbol

Rename the Kconfig symbol for all ITE SoC variants.
Rename the SOC_SERITES_ITE_IT8XXX2 to SOC_SERIES_IT8XXX2.

Signed-off-by: Tim Lin <tim2.lin@ite.corp-partner.google.com>

GTLin08 committed:
ITE: soc: Modify Kconfig default declare

Using the SOC_IT8XXX2_REG_SET_V2 instead of constantly adding new
variants of the IT82XX2 SoC.

Signed-off-by: Tim Lin <tim2.lin@ite.corp-partner.google.com>

aescolar commented on doc/releases/migration-guide-3.7.rst:
```diff
@@ -30,6 +30,10 @@ Boards
 
 * Reordered D1 and D0 in the `pro_micro` connector gpio-map for SparkFun Pro Micro RP2040 to match
   original Pro Micro definition. Out-of-tree shields must be updated to reflect this change.
+* ITE: Rename all SoC variant names, e.g., ``SOC_IT82202_AX`` is renamed to ``SOC_IT82202AX``.
```
Should this say?
```suggestion
* ITE: Rename all SoC variant names kconfig options, e.g., ``CONFIG_SOC_IT82202_AX`` is renamed to ``CONFIG_SOC_IT82202AX``.
```
if it is only the kconfig options and nothing else that changes (i.e. only those who have custom boards or happen to use those kconfig options in their code may realize)

GTLin08 replied:
Done.
